xt_training/metrics.py

```diff
@@ -247,6 +247,21 @@ def reset(self):
         self.latest_num_samples = 0
 
 
+class MultitoBinaryAUC():
+    """Convert the prediction from multi-class to binary, and calculate the AUC."""
+    def __init__(self):
+        self.roc_auc = ROC_AUC()
+
+    def __call__(self, y_pred, y):
+        y_pred_bi = torch.transpose(
+                        torch.vstack([y_pred[:, 0], torch.logical_not(y_pred[:, 0])]), 
+                        0, 1
+                    )
```

Comment:
`y_pred` is generally logits. As a result, shouldn't `torch.logical_not(y_pred[:, 0])` always return False for every row?

I would have thought that correct way to do this was to just do:

```python
y_pred = nn.functional.softmax(y_pred, dim=1)
y_pred_bi = torch.vstack(y_pred[:, 0], y_pred[:, 0:].sum(dim=1)).log()

y_bi = (y > 0).float()
```

Reply:
Thanks Tim! This was a stupid mistake. `y_pred` should be logits, and we need to make sure `y_pred_bi` is logits as the input of `ROC_AUC()`. Hence I changed the code following your advice:
```
y_pred = F.softmax(y_pred, dim=1)
y_pred_bi = torch.transpose(
    torch.vstack([y_pred[:, 0], y_pred[:, 1:].sum(dim=1)]).log(),
    0, 1
)
y_bi = (y > 0).float()
```